drivers/UARTSerial.cpp

```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#if DEVICE_SERIAL
+#if (DEVICE_SERIAL && DEVICE_INTERRUPTIN)
```

Comment:
Seems a bit funny to add this check here? Not seeing this using the InterruptIn APIs here, which usually what this guard is used for.

There currently isn't a target that has `SERIAL` but not `INTERRUPTIN` listed in its `device_has` descriptor, so this won't cause immediate issues. Just seems unnecessary.